### PR TITLE
store/mockstore: Use `Expression.Eval` to replace `EvalWithInnerCtx` for mockstore

### LINKED ISSUE #48770: Use `Expression.Eval` to replace `EvalWithInnerCtx` for mockstore

### PR DESCRIPTION
**Pull request by lcwangchao**
### What problem does this PR solve?

Issue Number: close #48770

Problem Summary:

After #48609 , we have added a `ctx` parameter to `Expression.Eval` to use context as an input instead of using the inner one. However, we left some `EvalWithInnerCtx` invokes when we could not find a context around the invoke. We should replace them with `Eval` in the later PRs.

### What is changed and how it works?

In this PR, we replace all the invokes of `EvalWithInnerCtx` in `store/mockstore` with `Eval` to use the context as a input parameter

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  >   

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: pkg/expression/distsql_builtin.go [hunk 1]
```diff
@@ -1109,10 +1105,10 @@ func newDistSQLFunctionBySig(sc *stmtctx.StatementContext, sigCode tipb.ScalarFu
 }
 
 // PBToExprs converts pb structures to expressions.
-func PBToExprs(pbExprs []*tipb.Expr, fieldTps []*types.FieldType, sc *stmtctx.StatementContext) ([]Expression, error) {
+func PBToExprs(pbExprs []*tipb.Expr, fieldTps []*types.FieldType, ctx sessionctx.Context) ([]Expression, error) {
```
Comment (hawkingrei) [inline pkg/expression/distsql_builtin.go:1108]: In general, we make ```ctx``` as a first parameter.

File: pkg/server/rpc_server.go [hunk 2]
```diff
@@ -239,6 +240,9 @@ func (s *rpcServer) createSession() (session.Session, error) {
 		action := &memory.PanicOnExceed{Killer: &vars.SQLKiller}
 		vars.MemTracker.SetActionOnExceed(action)
 	}
+	if err = vars.SetSystemVar(variable.MaxAllowedPacket, strconv.FormatUint(variable.DefMaxAllowedPacket, 10)); err != nil {
```
Comment (lcwangchao) [inline pkg/server/rpc_server.go:243]: Added this line to make sure the value `variable.DefMaxAllowedPacket` is present to make test pass. The old code is using `mock.Context` with `DefMaxAllowedPacket` set inner `ScalarFunction`. But we use the same sessioncontext object in this PR.

**Review by YangKeao — APPROVED** [review]
LGTM

**Review by tiancaiamao — APPROVED** [review]

**Review by XuHuaiyu — APPROVED** [review]

**Pull request merged**